### ISSUE TITLE
bcftools query %END prints inconsistent values depending on the full expression

### ISSUE THREAD

**Issue by tedyun** [samtools/bcftools#1147]
Hi,

I found this rather strange behavior in `bcftools query`. When using a gVCF with the INFO/END field, the `%END` query sometimes respect the values in the INFO field and sometimes it doesn't, depending on the expression. In the following two reproducible cases the only difference between the two is the additional `\t[%GT]` part in Case2 in the format, but you can see that in Case1 `%END` will just print the start position + 1, while in Case2 the actual INFO/END value is printed. 

Case1:
```
$ curl https://raw.githubusercontent.com/dnanexus-rnd/GLnexus/master/test/data/long_ref_intervals_A.gvcf | bcftools query -f"%CHROM\t%POS0\t%END\n" | head -5
```

Case2:
```
$ curl https://raw.githubusercontent.com/dnanexus-rnd/GLnexus/master/test/data/long_ref_intervals_A.gvcf | bcftools query -f"%CHROM\t%POS0\t%END\t[%GT]\n" | head -5
```

I'm using bcftools/htslib v1.9. I think the desired behavior here would be to always print the INFO/END field value, as it's a standard/reserved field defined in VCF specs v4.3. Thank you.

Best,
Ted

**Comment by garrettjstevens** [comment]
I think this was fixed in v1.10 (I'm not a bcftools developer, just a fellow user):

```
$ bcftools --version
bcftools 1.9
Using htslib 1.9
Copyright (C) 2018 Genome Research Ltd.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ bcftools query -f"%CHROM\t%POS0\t%END\n" https://raw.githubusercontent.com/dnanexus-rnd/GLnexus/master/test/data/long_ref_intervals_A.gvcf | head -5
21	1000	1001
21	1010	1015
21	1016	1017
21	2000	2004
21	2002	2007
```

```
$ bcftools --version
bcftools 1.10.2
Using htslib 1.10.2
Copyright (C) 2019 Genome Research Ltd.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ bcftools query -f"%CHROM\t%POS0\t%END\n" https://raw.githubusercontent.com/dnanexus-rnd/GLnexus/master/test/data/long_ref_intervals_A.gvcf | head -5
21	1000	1005
21	1010	1016
21	1016	2000
21	2000	2005
21	2002	2008
```

**Comment by tedyun** [comment]
Thank you Garrett! I confirmed that using bcftools v1.10.2 fixes this issue :)
